sycl/doc/extensions/proposed/sycl_ext_oneapi_address_cast.asciidoc

```diff
@@ -93,31 +93,64 @@ implementation supports.
 ----
 namespace sycl::ext::oneapi::experimental {
 
-template <access::address_space Space, access::decorated DecorateAddress,
+// Shorthands for address space names
+constexpr inline address_space global_space = sycl::access::address_space::global_space;
+constexpr inline address_space local_space = sycl::access::address_space::local_space;
+constexpr inline address_space private_space = sycl::access::address_space::private_space;
+constexpr inline address_space generic_space = sycl::access::address_space::generic_space;
+
+template <access::address_space Space,
           typename ElementType>
-multi_ptr<ElementType, Space, DecorateAddress>
+multi_ptr<ElementType, Space, access::decorated::no>
 static_address_cast(ElementType* ptr);
```

Comment:
out of curiosity, why removing the possibility to select the yes/no decoration ?

Reply:
The logic was that there shouldn't be two ways to do the same thing, because that's harder to teach.

Before this change, `*_address_cast` functions could be used to cast between address spaces _and_ to introduce decoration.  Now, they can only be used to cast between address spaces, which is aligned with their name.

`multi_ptr` already provides interfaces to move between decorated and undecorated pointers (via implicit casts), so we don't need to handle it here as well.